querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java

```diff
@@ -177,6 +181,19 @@ public DefaultConfiguration(
             }
         }
 
+        if (options.containsKey(QUERYDSL_VARIABLE_NAME_FUNCTION_CLASS)) {
+            try {
+                variableNameFunction = (Function<EntityType, String>) Class.forName(options.get(QUERYDSL_VARIABLE_NAME_FUNCTION_CLASS)).newInstance();
+            } catch (RuntimeException e) {
+                throw e;
+            } catch (Exception e) {
+               variableNameFunction = new DefaultVariableNameFunction();
```

Comment:
I couldn't find many example of logging, is this what you meant?
```
catch (RuntimeException e) {
   System.err.println(e.getMessage());
   processingEnv.getMessager().printMessage(Kind.ERROR, e.getMessage());
   throw e;
} catch (Exception e) {
   System.err.println(e.getMessage());
   processingEnv.getMessager().printMessage(Kind.WARNING, String.format("An exception prevented instantiating tha variable name function of type %s", options.get(QUERYDSL_VARIABLE_NAME_FUNCTION_CLASS)));
   variableNameFunction = new DefaultVariableNameFunction();
}
```
But this requires to adding `ProcessingEnvironment` to the `DefaultConfiguration` constructor.